transport/internal/tlsmux/listener.go

```diff
@@ -21,45 +21,130 @@
 package tlsmux
 
 import (
+	"context"
 	"crypto/tls"
+	"errors"
 	"net"
+	"sync"
+	"time"
+
+	"go.uber.org/net/metrics"
+	"go.uber.org/zap"
 )
 
+var (
+	errListenerClosed = errors.New("listener closed")
+
+	_sniffReadTimeout    = time.Second * 10
+	_tlsHandshakeTimeout = time.Second * 10
```

Comment:
curious to know on how did we come up with this timeout value?

Reply:
Bad rebase, I have added a comment above these constants. Will address in a separate pull request with real values taken from ERD